fix: I915205 - Include angus-mail runtime for jakarata.mail-api

raghavendran-gopalakrishnan opened the pull request:
https://internal.almoctane.com/ui/?p=131002/6001#/entity-navigation?entityType=work_item&id=915205

raghavendran-gopalakrishnan committed:
I915205: Include Angus-mail runtime

https://internal.almoctane.com/ui/?p=131002/6001#/entity-navigation?entityType=work_item&id=915205

buildmachine-sou-jenkins2 commented:
A developer build has not yet been created for this branch. Click [here](https://sou-jenkins2.swinfra.net/job/Create-New-Build/parambuild/?GITHUB_SERVER=github.com&GITHUB_ORGANISATION=Autoscaler&REPO_NAME=autoscaler&BRANCH_NAME=915205) to go ahead and create the build...

buildmachine-sou-jenkins2 commented:
CI Build Link:
https://sou-jenkins2.swinfra.net/job/Autoscaler/job/Autoscaler~autoscaler~915205~CI

raghavendran-gopalakrishnan committed:
Changing to jakarta.mail

To avoid..

2024-05-09T07:59:11.075618057Z [ERROR] Unused declared dependencies found:
2024-05-09T07:59:11.075656325Z [ERROR]    org.eclipse.angus:angus-mail:jar:2.0.3:compile

buildmachine-sou-jenkins2 commented:
The Documentation QA site for this branch has been built:
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/Autoscaler_CI_autoscaler_915205

raghavendran-gopalakrishnan committed:
Use runtime scope

andyreidz commented:
@rorytorneymf Sorry I can't get to this today, could you review?

rorytorneymf commented:
Is this a new convention we are using for PR titles? We normally just use:

I915205: Include angus-mail runtime for jakarata.mail-api

rorytorneymf commented on autoscale-email-alert-dispatcher/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Does this file need a new line?

rorytorneymf commented on autoscale-email-alert-dispatcher/src/test/java/com/github/autoscaler/email/alert/dispatcher/EmailDispatcherTest.java:
```diff
@@ -0,0 +1,39 @@
+        try {
+            EmailDispatcher emailDispatcher = new EmailDispatcher(emailDispatcherConfiguration);
+            emailDispatcher.dispatch("hello, this is a test email");
+        } catch (Exception e) {
+            Assert.fail("Error while trying to dispatch mail.");
+        }
```
```suggestion
        } catch (final Exception e) {
            Assert.fail("Error while trying to dispatch mail: " + e);
        }
```

rorytorneymf reviewed (approved):
Just a couple of small comments, thanks

raghavendran-gopalakrishnan commented:
> Is this a new convention we are using for PR titles? We normally just use:
> 
> I915205: Include angus-mail runtime for jakarata.mail-api

Yes just thought of incorporating https://www.conventionalcommits.org/en/v1.0.0/

raghavendran-gopalakrishnan committed:
review comments

rorytorneymf commented:
> > Is this a new convention we are using for PR titles? We normally just use:
> > I915205: Include angus-mail runtime for jakarata.mail-api
> 
> Yes just thought of incorporating https://www.conventionalcommits.org/en/v1.0.0/

Ok no worries, think its maybe better to stay consistent though with what everyone in the rest of the team is using. 

We should maybe discuss if everyone in the team should be using this new convention @dermot-hardy?

raghavendran-gopalakrishnan commented:
> > > Is this a new convention we are using for PR titles? We normally just use:
> > > I915205: Include angus-mail runtime for jakarata.mail-api
> > 
> > 
> > Yes just thought of incorporating https://www.conventionalcommits.org/en/v1.0.0/
> 
> Ok no worries, think its maybe better to stay consistent though with what everyone in the rest of the team is using.
> 
> We should maybe discuss if everyone in the team should be using this new convention @dermot-hardy?

I don't see any conventions used till now in Fusion, it has always been < TicketID > : < Title >.
Interestingly, UX-Aspects team follow this convention. It looks good. 
https://github.houston.softwaregrp.net/Inner-Source-Incubating/ux-aspects-universal/commits/master
